include/experimental/__p1673_bits/blas1_dot.hpp

```diff
@@ -100,10 +101,18 @@ Scalar dot(
                 v1.static_extent(0) == v2.static_extent(0));
 
   using size_type = std::common_type_t<SizeType1, SizeType2>;
-  for (size_type k = 0; k < v1.extent(0); ++k) {
-    init += v1(k) * v2(k);
-  }
-  return init;
+  using scalar_type = std::common_type_t<ElementType1, ElementType2, Scalar>;
+  using std::ranges::iota_view;
+  using std::ranges::begin;
+  using std::ranges::end;
+
+  iota_view range{size_type{}, v1.extent(0)};
+
+  Scalar sum = std::transform_reduce(exec, begin(range), end(range), init,
+    std::plus<scalar_type> {},
+    [=](size_type i) { return v1[i] * v2[i]; });
```

Comment:
This looks correct.  Another approach would be to use `std::views::transform` on the `iota` view to turn `i` into `v1[i]` (or `v2[i]`).  Then you could use `std::plus<void>` directly, instead of creating a lambda that captures `v1` and `v2`.

Reply:
Thanks for the input. Can you show me what that would look like?